### PR TITLE
09 feat: search results view 

### PR DESCRIPTION
**Pull request by paschalidi**
This ticket is about the https://jira.dhis2.org/browse/DHIS2-9187

### What is this PR about?
Basically this implements the design doc, the version I implemented during the time everybody was on vacation. Is pretty close to the final result but some things are missing. 
The missing things are:
1. the new card design - there is already a [ticket](https://jira.dhis2.org/browse/DHIS2-9239) for that and is implemented later on this chain (see 17)
2. the fallback functionality - https://jira.dhis2.org/browse/DHIS2-9244
3. the "View latest enrolment" button and its functionality - https://jira.dhis2.org/browse/DHIS2-9245
4. the pagination is still returning the total number of rows - https://jira.dhis2.org/browse/DHIS2-9246
5. the "Create new user" functionality - we need to create a design doc for this

I guess if we could review this and add the missing things separately in the end of the chain that would be ideal for me. But I am happy to discuss it with you. If you have any comments, thoughts or complains please dont hesitate :)

### CODE REVIEW

**Review by JoakimSM — COMMENTED** [review]
Looks quite good. Made some minor suggestions. Please have a look.

File: src/core_modules/capture-core/components/Pages/Search/SearchResults/SearchResults.component.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
+// @flow
+import React from 'react';
+import i18n from '@dhis2/d2-i18n';
+import withStyles from '@material-ui/core/styles/withStyles';
+import { Pagination } from 'capture-ui';
+import { Button } from '@dhis2/ui-core';
+import CardList from '../../../CardList/CardList.component';
+import withNavigation from '../../../Pagination/withDefaultNavigation';
+import { searchScopes } from '../SearchPage.container';
```
[the remark this answers is not in the document]
Comment (paschalidi) [reply src/core_modules/capture-core/components/Pages/Search/SearchResults/SearchResults.component.js:9]: This is happening in the 11th of this chain. Lets leave this comment open here to remember

**Comment by paschalidi** [comment]
Missclicked

File: src/core_modules/capture-core/components/Pages/Search/SearchResults/SearchResults.component.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            break;
+        case searchScopes.TRACKED_ENTITY_TYPE:
+            searchViaAttributesOnScopeTrackedEntityType({ trackedEntityTypeId: searchScopeId, formId, page: newPage });
+            break;
+        default:
+            break;
+        }
+    };
+
+    const viewTrackedEntityDashboard = ({ item: { id, tei: { orgUnit: orgUnitId } } }) => {
```
[the remark this answers is not in the document]
Comment (paschalidi) [reply src/core_modules/capture-core/components/Pages/Search/SearchResults/SearchResults.component.js]: I renamed to to look more like a react component. It looks to me it is more of a function react component that a function that returns an element. Basically the same thing but in react world slightly different convention in the naming.

File: src/core_modules/capture-core/components/Pages/Search/SearchResults/SearchResults.component.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    searchGroupForSelectedScope,
+    rowsCount,
+    rowsPerPage,
+    currentPage,
+    currentSearchScopeType,
+    currentSearchScopeId,
+    currentFormId,
+    currentSearchTerms,
+}: Props) => {
+    const handlePaginationChange = (searchScopeType, searchScopeId, formId, newPage) => {
```
[the remark this answers is not in the document]
Comment (paschalidi) [reply src/core_modules/capture-core/components/Pages/Search/SearchResults/SearchResults.component.js]: Good points Joakim thanks

**Comment by paschalidi** [comment]
@JoakimSM the comments have been addressed. Thanks for the review. Perhaps you can have another look when you got the time

File: src/core_modules/capture-core/components/Pages/Search/SearchResults/SearchResults.component.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (JoakimSM) [reply src/core_modules/capture-core/components/Pages/Search/SearchResults/SearchResults.component.js]: Ok, I like the changes 👍

**Review by JoakimSM — APPROVED** [review]
👍 

Reminder for a later PR in the chain: [searchScopes](https://github.com/dhis2/capture-app/pull/848/files#r483478183)

**Pull request merged**